pandas/tseries/index.py

```diff
@@ -762,6 +762,8 @@ def astype(self, dtype):
             return self.asi8.copy()
         elif dtype == _NS_DTYPE and self.tz is not None:
             return self.tz_convert('UTC').tz_localize(None)
+        elif dtype == str:
+            return Index(self.format(), dtype=object)
```

Comment:
Or `self._shallow_copy(values=self.format(), infer=True)` should cover all the attributes such as `name`